pyuploadtool/changelog/changelog.py

```diff
@@ -0,0 +1,106 @@
+
+import os
+
+
+# order them by priority
+# https://cheatography.com/albelop/cheat-sheets/conventional-commits/
+CONVENTIONAL_COMMIT_SPEC = {
+    "feat":     "Features",
+    "fix":      "Bug Fixes",
+    "perf":     "Performance Improvements",
+    "docs":     "Documentation",
+    "ci":       "Continuous Integration",
+    "refactor": "Refactoring",
+    "test":     "Tests",
+    "build":    "Builds",
+    "revert":   "Reverts",
+    "chore":    "Chores",
+    "others":   "Commits"
+}
+
+
+class Changelog:
+    def __init__(self, title=os.getenv("CHANGELOG_TITLE"), commit_link_prefix=None):
+        """
+        Generates a changelog by arranging the commits according
+        to the Conventional Commit Spec
+
+        :param title: the title of the release, generally, the tag name
+        :type title: str
+
+        :param commit_link_prefix: a link prefix, which can be used to show a commit
+        for example
+        commit_link_prefix = https://github.com/$GITHUB_REPOSITORY/commit
+        here, we will add the commit hash to the end.
+        :type commit_link_prefix: str
+        """
+
+        # remove the trailing hash
+        self.commit_link_prefix = commit_link_prefix.rstrip("/")
+        self.title = title
+
+        self._data = dict()
+
+        for spec in CONVENTIONAL_COMMIT_SPEC:
+            self._data[spec] = list()
+
+    def push(self, commit: dict):
+        """
+        Adds a commit to the changelog and aligns each commit
+        based on their category. See CONVENTIONAL_COMMIT_SPEC
+        :return: The classification of the commit == CONVENTIONAL_COMMIT_SPEC.keys()
+        :rtype: str
+        """
+        message = commit.pop("message")
```

Comment:
You should insert more blank lines IMO. I insert blank lines after every docstring to separate it visually from the actual code. Also, before and after loops, I like to have blank lines for the same reason.